pkg/gofr/datasource/sql/query_builder_test.go

```diff
@@ -0,0 +1,154 @@
+package sql
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func Test_InsertQuery(t *testing.T) {
+	tableName := "user"
+	fieldNames := []string{"id", "name"}
+
+	tests := []struct {
+		dialect  string
+		expected string
+	}{
+		{
+			dialect:  "mysql",
+			expected: "INSERT INTO `user` (`id`, `name`) VALUES (?, ?)",
+		},
+		{
+			dialect:  "postgres",
+			expected: `INSERT INTO "user" ("id", "name") VALUES ($1, $2)`,
+		},
+		{
+			dialect:  "any-other-dialect",
```

Comment:
This test case with dialect as `any-other-dialect` don't make a lot of sense here. As the quote function also works only on two dialects and we also don't have a default case here:

```golang
func quote(dialect string) string {
	if DialectPostgres == dialect {
		return QuoteDouble
	}

	return QuoteBack
}
```

It's better to remove this test case or if we really want to include it we should return an error in case of un-supported dialect but that i think should be handled much before code coming here.